src/main/java/com/github/pedrovgs/androidwifiadb/view/AndroidDevices.java

```diff
@@ -0,0 +1,170 @@
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.openapi.wm.ToolWindowFactory;
+import com.intellij.ui.content.Content;
+import com.intellij.ui.content.ContentFactory;
+
+import javax.swing.JPanel;
+
+/**
+ * Created by vgaidarji on 10/23/15.
+ */
+public class AndroidDevices implements ToolWindowFactory, View, DeviceAction {
+
+    private int INTERVAL_REFRESH_DEVICES = 1000;
+    private static final String ANDROID_WIFI_ADB_TITLE = "Android WiFi ADB";
+    private static final NotificationGroup NOTIFICATION_GROUP =
+            NotificationGroup.balloonGroup(ANDROID_WIFI_ADB_TITLE);
+
+    private final AndroidWiFiADB androidWifiADB;
+    private JPanel toolWindowContent;
+    private final CardLayoutDevices cardLayoutDevices;
+
+    public AndroidDevices() {
+        CommandLine commandLine = new CommandLine();
+        ADBParser adbParser = new ADBParser();
+        ADB adb = new ADB(commandLine, adbParser);
+        this.androidWifiADB = new AndroidWiFiADB(adb, this);
+        cardLayoutDevices = new CardLayoutDevices(toolWindowContent, this);
+    }
+
+    @Override
+    public void createToolWindowContent(Project project, ToolWindow toolWindow) {
+        this.androidWifiADB.updateProject(project);
+
+        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
+        Content content = contentFactory.createContent(toolWindowContent, "", false);
+        toolWindow.getContentManager().addContent(content);
+
+        ApplicationManager.getApplication().executeOnPooledThread(new Runnable() {
+            public void run() {
+                setupUI();
+                monitorDevices();
+            }
+        });
+    }
+
+    private void setupUI() {
+        ApplicationManager.getApplication().invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                cardLayoutDevices.createAndShowGUI();
+            }
+        });
+    }
+
+    /**
+     * Monitors connected devices state and triggers UI updates.
+     */
+    private void monitorDevices() {
+        ApplicationManager.getApplication().executeOnPooledThread(new Runnable() {
+            public void run() {
+                while (true) {
+                    boolean refreshRequired = androidWifiADB.refreshDevicesList();
+                    if (refreshRequired) {
+                        updateUi();
+                    }
+
+                    try {
+                        Thread.sleep(INTERVAL_REFRESH_DEVICES);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public void showNoConnectedDevicesNotification() {
+        showNotification(ANDROID_WIFI_ADB_TITLE,
+                "There are no devices connected. Review your USB connection and try again. ",
```

Comment:
The message was copied from [here](https://github.com/pedrovgs/AndroidWiFiADB/blob/master/src/main/java/com/github/pedrovgs/androidwifiadb/action/AndroidWiFiADBAction.java#L59). I've removed the additional space from both places.